### PR TITLE
Run3-alca202 Add a new AlCaProducer for HBHE muons with reduced data volume on o/p

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add a new AlCaProducer for HBHE muons with reduced data volume on o/p

#### PR validation:

Tested against a Reco o/p

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+// constructors and destructor
+//
+AlCaHcalHBHEMuonFilter::AlCaHcalHBHEMuonFilter(edm::ParameterSet const& iConfig)
+    : prescale_(iConfig.getParameter<int>("prescale")),
+      labelHBHEMuonVar_(iConfig.getParameter<edm::InputTag>("hbheMuonLabel")),
+      tokHBHEMuonVar_(consumes<HcalHBHEMuonVariablesCollection>(labelHBHEMuonVar_)) {
+  edm::LogVerbatim("HBHEMuon") << "Parameters read from config file \n\t prescale_ " << prescale_ << "\n\t Labels "
+                               << labelHBHEMuonVar_;
+}  // AlCaHcalHBHEMuonFilter::AlCaHcalHBHEMuonFilter  constructor
+
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc:59]: isn't the destructor missing here?

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonFilterProducer_cfg.py [hunk 2]
```diff
@@ -33,9 +38,11 @@
         ),
                                                          eventAutoFlushCompressedSize = cms.untracked.int32(5242880),
                                                          outputCommands = process.OutALCARECOHcalCalHBHEMuonFilter.outputCommands,
-                                                         fileName = cms.untracked.string('PoolOutput.root'),
+                                                         fileName = cms.untracked.string('OutputHBHEMuon.root'),
                                       )
 
+process.AlcaHBHEMuonFilter.Triggers = []
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonFilterProducer_cfg.py:44]: I'm puzzled by this line, should sime triggers be in here? Or better: shouldnt this be governed by the trigger bit, something like https://github.com/cms-sw/cmssw/blob/master/Calibration/TkAlCaRecoProducers/python/ALCARECOSiPixelCalCosmics_cff.py#L19 and then an update to the GT?

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+process.load('Configuration.StandardSequences.Services_cff')
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.load("Configuration.StandardSequences.GeometryRecoDB_cff")
+process.load("Configuration.StandardSequences.MagneticField_cff")
+process.load('Configuration.StandardSequences.AlCaRecoStreams_cff')
+process.load('Configuration.StandardSequences.EndOfProcess_cff')
+process.load('Configuration.StandardSequences.FrontierConditions_GlobalTag_cff')
+from Configuration.AlCa.autoCond import autoCond
+process.GlobalTag.globaltag=autoCond['run2_mc']
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py:14]: `run2_mc` is for 2016 MC, is this intended?

Comment (tvami) [reply Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py:14]: the rest of the config is for HLTPhysics data, right?

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                                                         dataset = cms.untracked.PSet(
+        dataTier = cms.untracked.string('ALCARECO'),
+        filterName = cms.untracked.string('ALCARECOHcalCalHBHEMuonProducerFilter')
+        ),
+                                                         eventAutoFlushCompressedSize = cms.untracked.int32(5242880),
+                                                         outputCommands = process.OutALCARECOHcalCalHBHEMuonProducerFilter.outputCommands,
+                                                         fileName = cms.untracked.string('OutputBHEMuonProducerFilter.root'),
+                                      )
+
+process.alcaHcalHBHEMuonProducer.triggers = []
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py:45]: See my comment earlier about this

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc [hunk 5]
```diff
@@ -0,0 +1,903 @@
...
+  }
+#endif
+
+  bool changed = true;
+  bool flag = hltConfig_.init(iRun, iSetup, processName_, changed);
+  edm::LogVerbatim("HBHEMuon") << "Run[" << nRun_ << "] " << iRun.run() << " hltconfig.init " << flag << std::endl;
+
+  theHBHETopology_ = &iSetup.getData(tok_htopo_);
+  const HcalRespCorrs* resp = &iSetup.getData(tok_respcorr_);
+  respCorrs_ = new HcalRespCorrs(*resp);
```
Comment (tvami) [inline Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc:781]: `new` is discouraged, use `std::unique_ptr` instead

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc (hunk 1 above)
Comment (makortel) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc:60]: No, the destructor is defined in
https://github.com/cms-sw/cmssw/blob/76ab0ba0ab44a1c8ee28be749d3db8aade2c987d/Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc#L35

Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc:60]: Added. I was relying on the default one which gets automatically added

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonFilterProducer_cfg.py (hunk 2 above)
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonFilterProducer_cfg.py:45]: For tests I am not testing on triggers. Production jobs relies on them

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py (hunk 3 above)
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py]: Corrected. Here we do this for HLTPhysics - but it is just an example

File: Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py (hunk 4 above)
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/test/AlCaHBHEMuonProducerFilter_cfg.py:45]: Same here - only for test

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc [hunk 6]
```diff
@@ -0,0 +1,903 @@
...
+  edm::LogVerbatim("HBHEMuon") << "Flags used: UseRaw " << useRaw_ << " CollapseDepth " << collapseDepth_ << ":"
+                               << mergedDepth_ << " IsItPlan1 " << isItPlan1_ << " IsItPreRecHit " << isItPreRecHit_
+                               << " UseMyCorr " << useMyCorr_;
+
+  //create the objects for HcalHBHEMuonVariables which has information of isolated muons
+  produces<HcalHBHEMuonVariablesCollection>(labelHBHEMuon_);
+  edm::LogVerbatim("HcalIsoTrack") << " Expected to produce the collections:\n"
+                                   << "HcalHBHEMuonVariablesCollection with label " << labelHBHEMuon_;
+}
+
```
[the remark this answers is not in the document]
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc:229]: added

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc (hunk 5 above)
Comment (bsunanda) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonProducer.cc]: taken care of

File: Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc (hunk 1 above)
Comment (tvami) [reply Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalHBHEMuonFilter.cc:60]: @makortel I think we looked at the file at a different point in time, now it's indeed there

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by tvami** [comment]
@bsunanda let's squash and then sign it

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca

**Comment by bsunanda** [comment]
@qliphy @perrotta Can you merge this PR?

**Comment by qliphy** [comment]
+1

**Pull request merged**